### ISSUE TITLE
Type casting causes used imports to be removed

### ISSUE THREAD

**Issue by zbarbuto** [buehler/typescript-hero#301]
#### System information

- OS: OSX
- TypeScript Hero version: 1.5.0
- VSCode version: 1.15.1

#### Description

I've enabled the new format on save option, however, it seems that whenever I save a file, any imports used purely for typing are stripped out.

##### Steps to reproduce

1. In a typescript file import an interface and use it just for a type:

```typescript
import { IFoo } from '../interfaces/foo';
function bar(arg: IFoo) {
  // do some stuff
}
```

2. Save the document
3. The import is removed resulting in typescript errors

**Comment by buehler** [comment]
Hmm strange.. I tested it and in my local version it works like a charm :-(

**Comment by zbarbuto** [comment]
@buehler If it helps I've made a couple of simple test cases that reproduce my issue:

1:
index.ts
```typescript
import { foo, bar } from './lib/foo' // both get removed (or just foo if imported) on organize

let bar = <any>foo();
```

lib/foo.ts
```typescript
export function foo() { return 'bar' }
```

2.

index.ts
``` typescript
import { foo, bar } from './lib/test'; // bar gets removed on organize

export const baz = <any>foo();
export const bing = {
  foo: bar()
}
```

lib/foo.ts
```typescript
export function foo() { return 'bar' }
export function bar() { return 'baz' }

```


The fact that only bar gets removed in the second case but foo gets removed in the first leads me to think it's an issue related to type casting.

**Comment by zbarbuto** [comment]
Confirmed to be a typecasting related issue since as soon as I removed it from the few files I was having issues with the problem went away.

**Comment by buehler** [comment]
@zbarbuto thanks for the research!
Hmm that actually could be the case. This sounds more of an issue for the node parser:
https://github.com/TypeScript-Heroes/node-typescript-parser

I'll have a look at it

**Comment by paulussup** [comment]
Not so heroic today @zbarbuto

**Comment by adrianksiezarczyk** [comment]
Is there any solution to for example disable auto-removing imports?

**Comment by zbarbuto** [comment]
Until it's fixed set to: `"typescriptHero.resolver.organizeOnSave": false`

**Comment by buehler** [comment]
Fixed by #327
